cvat-canvas/src/typescript/canvasModel.ts

```diff
@@ -608,6 +611,8 @@ export class CanvasModelImpl extends MasterImpl implements CanvasModel {
                     } else {
                         this.data.exception = new Error('Unknown error occured when fetching image data');
                     }
+                    this.data.objects = prevObjects;
+                    this.data.zLayer = prevZLayer;
```

Comment:
Perhaps it should be:

```js
if (this.data.objects === objectStates && this.data.zLayer === zLayer) {
     this.data.objects = prevObjects;
     this.data.zLayer = prevZLayer;
}
```

Otherwise these values may be not relevant to the moment when the promise is already rejected (one more `setup` has been called in the time range).

Reply:
Also let's add a comment with explanation to the code